### ISSUE TITLE
[Bug] Everytime app-restart/Ha-restart the router/nodes is as new device

### ISSUE THREAD

**Issue by MariusStensrod** [Vaskivskyi/ha-asusrouter#886]
### The problem

Everytime I restart the app or restart HA, the router and mesh-nodes is as a new device in HA. I then have to designate a room and a new name. As the name always is "ZenWiFi AX". This breaks my flows. Are you able to fix this?

### Your device model

ZenWiFi AX XT8

### Firmware type

Stock

### Firmware version

3.0.0.4.388

### Integration version with the issue

v0.34.1

### Method of the integration installation

HACS

### What version and type of Home Assistant installation do you use

 Core 2024.11.2

**Comment by mrZeffka** [comment]
I confirm, mine are constantly being renamed too. And this mess is also annoying, when I have two routers and all the values ​​are dumped into one device.

**Comment by SebastiaanKlippert** [comment]
@mrZeffka Do you also use XT8? Not sure if this is only related to the XT8.

I have a similar issue with my 2 XT8's. It adds a new device sometimes, but unsure when.
It currently looks like this and it is annoying that the controls move to another device, that means updating the dashboards and automations again.

![Image](https://github.com/user-attachments/assets/90eb22a2-54ba-4397-bf2a-6e903465b429)

**Comment by mrZeffka** [comment]
@SebastiaanKlippert Yes, I have exactly the same situation as you. At the moment I stopped using this integration. I have AC86U.

**Comment by Vaskivskyi** [comment]
Hello,

In all the cases, one device is a router and one - an AiMesh node? Or both AiMesh nodes?

@SebastiaanKlippert, total of what you have - are 2 XT8, correct? Buth showing up as both ZenWiFi AX and RT-AX95Q?

**Comment by MariusStensrod** [comment]
Hello,

I have a XT8 gen1 as a router 
-XT8 gen2 hardwired AiMesh node 
-ExpertWiFi EBA63 hardwired AiMesh node 
-XT8 gen1 WiFi AiMesh node.

Running latest stable FW on ASUS router and nodes.
Running v 0.35.0 of this integration.
Running latest stable HAOS releases:

![Image](https://github.com/user-attachments/assets/5b4a833a-0696-494e-b82f-2b47a17c56bc)


Now it only happens with the router device. I am not sure when it changed, but my AiMesh nodes does not add as a new device anymore.  Only the router device.

**Comment by SebastiaanKlippert** [comment]
Hi @Vaskivskyi Yes correct.
I have two XT8. One in AP mode and then one as AiMesh node. 

In HASS all the devices and sensors for the main node are listed in one of the 4 "ZenWiFi AX" devices.
![Image](https://github.com/user-attachments/assets/79493c56-e790-4dbb-a13f-0cdc83caa74e)

Currently they don't work, all unavailable (might be a different issue).

And all the devices and controls for the AiMeh node are under one of the "RT-AX95Q" nodes

![Image](https://github.com/user-attachments/assets/46662137-2710-45d5-a183-2bf4e9a7a2a8)

I also see now that the firmware version for the node is not correct. According to the ASUS device web interface both are up-to-date.

![Image](https://github.com/user-attachments/assets/dda323e2-de23-436f-85c1-50ee4226442d)
All update intervals and cache settings are on default.

**Comment by SebastiaanKlippert** [comment]
Just an update to show you how random it seems to me. 
After update to HASS 2025.5.0 my list looks like this.

![Image](https://github.com/user-attachments/assets/530a88b8-4efa-4acb-a7be-32ea2b7c5e89)

Seems like two ZenWiFi AX changed to RT-AX95Q. More annoyingly, all my automations in my dashboards and automations are now also gone, they are now on a different device.

To be fair there was also a FW Upgrade last week, but it all works fine until HASS restarts.

**Comment by Vaskivskyi** [comment]
If the issue is happening again on integration version `0.40.0`, please open a new issue. At least one bug influencing multi-device setups has been fixed, so new data will be needed for any troubleshooting
